### ISSUE TITLE
AI model returning commented code

### ISSUE THREAD

**Issue by endocytosis** [stackblitz/bolt.new#2862]
**Issue:** Upon making changes to a file, the AI model returns commented sections (e.g. /* previous imports remain the same */) instead of the complete code. Tracking similar issues here.

**Comment by Borteist** [comment]
Will this issue ever be solved?

**Comment by AlexandreCalvet** [comment]
I will not renew my subscription. This bug is becoming more and more frequente and is breaking my code all the time. 
Is there a work around to force the inclusion of the code to keep ?

**Comment by Plactrix** [comment]
Not only does this issue still persist, it's also costed our team, and I assume a lot of others, a lot of money. Despite us asking it to fix it, it usually takes upwards of 3 tries before it actually properly corrects it. One message costs us over 100k credits.

We will also not be renewing until this issue is addressed.

**Comment by zfr19389** [comment]
Even I am facing the same issue. Lost more than 100k credits in getting the entire content of file but everytime it keeps on giving the comments instead of actual code. Something as shown below:

);
};
</boltArtifact>

This implementation:

1. Creates a dedicated workflow page that handles both document upload and initial analysis
2. Uses the useInitialAnalysis hook to trigger the analysis after workflow completion
3. Automatically navigates to the chat page when analysis is complete
4. Preserves messages and session state across navigation
5. Shows loading and error states during analysis

The workflow is now:
1. User starts on workflow page
2. Uploads documents and configures settings
3. Clicks Generate/Submit
4. Initial analysis is triggered automatically
5. User is redirected to chat with initial analysis results
6. Chat continues with the same se

**Comment by nizaeros** [comment]
@endocytosis 
This issue has not fixed. ai is continously writing his modification to the chat conversation instead of directly modifying to the code pages. and also breaking the code with partial edits. 

This is exhausting all my token and time, important point is that i am at the middle of a project delivery and ths is making the situation crazy. please help

**Comment by swcrazyfan** [comment]
I'm having this same issue. Sometimes, I can fix it by every message sending the instruction to give full and complete code without placeholders, but I just switched over to editing files with Claude directly. Bolt.new is useless if it randomly does this.

**Comment by romainboyer** [comment]
While I had this more and more, the **best way I found to not have this problem was to ask Bolt to generate very small and specialized files.** This happen everytime the file is too long.

Problem is Bolt wants everytime to regenerate the complete file.
This causes 2 main problems:
- it thinks again the logic and some time you have unexpected regressions on code parts that were NOT concerned by the prompt
- it become lazy on some parts and forget to work on some things 

and the whole things depends on the context it can afford to maintain.

I left Bolt after losing hundreds of thousands token because of that and other things (netlify production configuration lost, typescript unfixable errors while pushing to production,...). The solution is not ready for production.

**Comment by quallakids** [comment]
I’ve been losing all my credits trying to resolve issues that are always returned as a text, claiming a solution, but the problem remains unsolved in the actual code. It’s incredibly frustrating to watch my credits being spent while the system fails to provide a real solution. This has happened with even the simplest tasks, such as moving a menu option to a new position.

**Comment by caitlinrouille** [comment]
Agreed. Having same issue and its costing me lots of money. I may cancel my subscription until fixed. I also don’t see how this would be treated as an enhancement when customers are losing money, if anything its a bug that should have been fixed before having a pricing model and kept in beta.

**Comment by codewithtyler** [comment]
@caitlinrouille, I understand why you're frustrated. To clarify on the labeling system in the this repo, The StackBlitz team uses 'enhancement' labels to track issues internally, which means they'll be used for public updates instead of addressing each one individually (as there are over 3,000).

**Comment by quallakids** [comment]
Thank you very much for your communication. Are you going to give me back
some compensation for the losen tokens?


Imatge inserida


Carles Porta
CEO
Qualla Systems SL
+34 607 31 54 18
 ***@***.***

**Comment by codewithtyler** [comment]
@quallakids you'll need to contact the Customer Support team if you want compensation for lost tokens as a result. You can do that by emailing support@stackblitz.com. Note, you should get an auto reply email. If you don't get one then that means they did not receive your email and will be unable to respond to you.

**Comment by Plactrix** [comment]
> With today's update this should now be solved. If you were experiencing this issue, Bolt should no longer be lazy. If you do continue to experience this issue please let us know.
> 
> https://x.com/boltdotnew/status/1881731948051415059

Unfortunately this still appears to exist (see attached + https://github.com/stackblitz/bolt.new/issues/5554)

![Image](https://github.com/user-attachments/assets/90e8ead3-1ef9-4eef-a899-56d8287393d0)

**Comment by Nemikolh** [comment]
@Plactrix Do you have a version of the file before the failure and right after?

**Comment by Plactrix** [comment]
> [@Plactrix](https://github.com/Plactrix) Do you have a version of the file before the failure and right after?

Of that one specifically, I do not unfortunately. I tried just now and it seems like it goes through fixes it afterward now.

When I was attempting that a couple of days ago, it wouldn't actually save the file after the prompt finished- it just kept reverting to the old file as if it didn't have permission to save the new one or something. That's another bug I occasionally run into, but I don't mind that as it is easy to get around. I took that screenshot midway through Bolt writing the new file. I suspect that's why it did not do the fixing process?

Also, does this fixing process use credits on top of the original prompt, or is done it at Bolt's expense?

**Comment by Plactrix** [comment]
Updating this again as I came across a similar issue: https://bolt.new/~/stackblitz-starters-sdeg5nt1

![Image](https://github.com/user-attachments/assets/6412ee0b-2777-4088-a2ed-b54d5ad41143)

Bolt created 3 pages with that text, and costed me 100k credits for that

**Comment by Jakoobster** [comment]
same i bought 10m to literally loose 1m just to this during my time on top of that you use them all in one go and the token cost is expensive off the rip this needs to be fixed
